### PR TITLE
Add resources to SumoCloud Siem Integration: 

### PR DESCRIPTION
**Pull request by duchatran**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
We add the following resources to the integration: 
- Classifier
- Mapper
- New Incident Type
- Several Incident Fields
See Release Notes for more details 

## Screenshots
- New Layout with a dedicated tab: 

image
- New Incident Type:
image

- New Incident Fields:

image

## Minimum version of Cortex XSOAR
- [x ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Thank you for your contribution!
Please take a look at my comments. Let me know if you need any help in addressing it.

Also, please note your tests are failing due to these changes. Please fix your test to support the manipulation you added.

**Comment by kgal-pan** [comment]
@lmilby-sumo - Please review unit tests as seems that [3 of them are failing assertions](https://app.circleci.com/pipelines/github/demisto/content/168926/workflows/65c44c7e-8dc9-4895-99e0-c6929cf874b7/jobs/528404).

In addition, please review the validation errors for [missing incident fields in mapper](https://app.circleci.com/pipelines/github/demisto/content/168926/workflows/65c44c7e-8dc9-4895-99e0-c6929cf874b7/jobs/528405?invite=true#step-109-92) and run `demisto-sdk update-release-notes` to add mapper, classifier and integration changes to release notes.

**Review by idovandijk — COMMENTED** [review]
Hi, thanks for the contribution. I have a few questions and comments:
1. Looks like the classifier is missing from the release notes, please add it.
2. I have an issue with the mapper locally, appears empty, not sure if it's just me:
image
can you check that it's ok if you upload the mapper from the PR to your instance?
3. I will check about the new content whether it should have `fromversion: 6.0.0`, we might need to change that to 6.1.0 or 6.2.0

Finally can we schedule a short demo, about 30 min, to see fetched incidents, layout, etc? It is difficult to understand the data manipulation with the classifier without seeing incidents for example
image

Let me know some time slots that could work, or  you could upload short GIFs showing examples instead if you prefer. Thanks!

**Comment by kgal-pan** [comment]
@duchatran - Any updates or assistance needed?

**Comment by duchatran** [comment]
Sorry, back from PTO now. Let me work on these

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py [hunk 1]
```diff
@@ -94,6 +94,14 @@ def add_list_to_q(q, fields, args):
                 q = add_to_query(q) + '{}:"{}"'.format(arg_field, arg_value)
     return q
 
+def convert_api_url_to_service_url(api_url):
+    '''
+    '''
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py]: Removed this function

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py [hunk 2]
```diff
@@ -94,6 +94,14 @@ def add_list_to_q(q, fields, args):
...
+    if (api_url.rfind('stag')==-1):
+        return api_url[:-4].replace('api','service')
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py]: I removed this function as it's not used

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py [hunk 3]
```diff
@@ -94,6 +94,14 @@ def add_list_to_q(q, fields, args):
...
+    else:
+        api_url
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py]: I removed this function as it's not used

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py [hunk 4]
```diff
@@ -239,10 +254,12 @@ def insight_get_details(client: Client, args: Dict[str, Any]) -> CommandResults:
 
     resp_json = client.req('GET', 'sec/v1/insights/{}'.format(insight_id), query)
     insight = insight_signal_to_readable(resp_json)
+    insight['SumoUrl'] = craft_sumo_url(demisto.getParam('instance_endpoint'),'insight',insight_id)
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py]: Fixed

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py [hunk 5]
```diff
@@ -283,9 +300,10 @@ def signal_get_details(client: Client, args: Dict[str, Any]) -> CommandResults:
     signal = client.req('GET', 'sec/v1/signals/{}'.format(signal_id))
     signal.pop('allRecords', None)  # don't need to display records from signal
     signal = insight_signal_to_readable(signal)
+    signal['SumoUrl'] = craft_sumo_url(demisto.getParam('instance_endpoint'),'signal',signal_id)
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py:306]: I moved this param to the client object.

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.yml [hunk 6]
```diff
@@ -2,7 +2,7 @@ commonfields:
   id: SumoLogicSEC
   version: -1
 name: SumoLogicSEC
-display: Sumo Logic Cloud SIEM
+display: Sumo Logic Cloud SIEM Dev
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.yml]: Thanks, fixed

File: Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+- **Sumo Logic Cloud SIEM Insight URL** field contains the URL of the Insight on the Cloud SIEM side
+
+###### Layouts
+1. Sumo Logic Insight
+- New layout for Sumo Logic Insights (Available from Cortex XSOAR 6.0.0).
+
+###### Mappers
+1. Sumo Logic Insight - Incoming Mapper
+- New incoming mapper for Sumo Logic Insight (Available from Cortex XSOAR 6.0.0).
+
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md]: Fixed (manually as demisto update-release-notes doesn't detect any change for me)

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py [hunk 8]
```diff
@@ -683,6 +701,7 @@ def fetch_incidents(client: Client, max_results: int, last_run: Dict[str, int],
         query['recordSummaryFields'] = record_summary_fields
     incidents = []
     hasNextPage = True
+    instance_endpoint = demisto.getParam('instance_endpoint')
```
[the remark this answers is not in the document]
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.py]: Fixed

**Comment by kgal-pan** [comment]
Hi @duchatran - 

- Looks like some unit tests are failing because of `AssertionError`s and `AttributeError`. Please [review](https://app.circleci.com/pipelines/github/demisto/content/172922/workflows/25fe3d38-e6d4-44ad-8330-c2f61d7c8ca1/jobs/535725) and fix them.
- The mapper mentions mappings for fields that do not pair with the fields added in the Pack. For example, the mapper specifies mapping for field named `Insight Entity` but the actual field name is `Sumo Logic Cloud SIEM Insight Entity` in file [Packs/SumoLogic_Cloud_SIEM/IncidentFields/incidentfield-Insight_Entity.json](https://github.com/demisto/content/pull/19448/files#diff-d56a5d81e5de6a40b607ddf98b60b17c0efdb7f4bd985a7dd51c17c049f71b30).
- Please update the release notes to include the mapper and classifier but running `demisto-sdk update-release-notes Packs/SumoLogic_Cloud_SIEM`

File: Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+- **Sumo Logic Cloud SIEM Insight URL** field contains the URL of the Insight on the Cloud SIEM side
+
+#### Layouts
+##### Sumo Logic Insight
+- New layout for Sumo Logic Insights (Available from Cortex XSOAR 6.2.0).
+
+#### Mappers
+##### Sumo Logic Insight - Incoming Mapper
+- New incoming mapper for Sumo Logic Insight (Available from Cortex XSOAR 6.2.0).
+
```
Comment (JasBeilin) [inline Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md:26]: No release note entry was found for the mapper "Sumo Logic Insight - Mapper" in the SumoLogic_Cloud_SIEM pack. This happens cause the RN contains "Incoming Mapper" instead. @idovandijk What do you think? Should we change to Mapper?

No release note entry was found for the classifier "Sumo Logic Insight - Classifier" and
 Please rerun the update-release-notes command without -u to generate an updated template. If you are trying to exclude an item from the release notes, please refer to the documentation found here - https://xsoar.pan.dev/docs/integrations/changelog#excluding-items

Comment (idovandijk) [reply Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md:26]: I think the RN should generate what the name of the mapper is. They could choose to change the name of the mapper to Incoming Mapper, but then I believe the SDK would change it to `Sumo Logic Insight - Incoming Mapper - Incoming Mapper`, wouldn't it?

File: Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+						"filters": [],
+						"root": "entity",
+						"transformers": [
+							{
+								"operator": "ConvertKeysToTableFieldFormat"
+							}
+						]
+					}
+				},
+				"Insight Involved Entities": {
```
Comment (kgal-pan) [inline Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json:23]: Please update to `Sumo Logic Cloud SIEM Insight Involved Entities`

File: Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+						"filters": [],
+						"root": "involvedEntities",
+						"transformers": [
+							{
+								"operator": "ConvertKeysToTableFieldFormat"
+							}
+						]
+					}
+				},
+				"Insight Signals": {
```
Comment (kgal-pan) [inline Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json:34]: Please update to `Sumo Logic Cloud SIEM Insight Signals`.

File: Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json (hunk 10 above)
Comment (kgal-pan) [inline Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json:23]:
```suggestion
				"Sumo Logic Cloud SIEM Insight Involved Entities": {
```

File: Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json (hunk 11 above)
Comment (kgal-pan) [inline Packs/SumoLogic_Cloud_SIEM/Classifiers/classifier-mapper-incoming-Sumo_Logic_Cloud_SIEM.json:34]:
```suggestion
				"Sumo Logic Cloud SIEM Insight Signals": {
```

File: Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md (hunk 9 above)
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md:26]: @idovandijk : the SDK uses "Sumo Logic Insight - Mapper", should I keep it?

Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md:26]: @JasBeilin I kept having issues with the update-release-notes command until today. It seems paths are particularly  important for some demisto-sdk commands. Having a specific example in online doc page would be great (e.g https://xsoar.pan.dev/docs/documentation/release-notes)

**Comment by duchatran** [comment]
@kgal-pan Also, I don't have the Partner ID for the Contribution Form.

File: Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md (hunk 9 above)
Comment (duchatran) [reply Packs/SumoLogic_Cloud_SIEM/ReleaseNotes/1_1_5.md:28]: Updated the RN

**Comment by duchatran** [comment]
@kgal-pan Thanks, submitted Contribution Form. Let me know what else you need.

**Review by JasBeilin — APPROVED** [review]

**Comment by duchatran** [comment]
Thanks for the review today @kgal-pan @idovandijk @JasBeilin. I have added the indicator extraction rule to the Insight type.

**Comment by JasBeilin** [comment]
@duchatran, please merge from our master in both you main branch and contribution branch, as the validation test is failing.
Merge from masterRun ./Utils/git_pull_master_into_fork.sh or merge manually from upstream demisto content.

**Comment by JasBeilin** [comment]
@duchatran 
Great, just last finishes and we are good to go!
1. The name attribute of classifier-Sumo_Logic_Cloud_SIEM.json (currently Sumo Logic Insight - Classifier) should be identical to its `id` attribute (SumoLogicInsight)
2. Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogicCloudSIEM/SumoLogicCloudSIEM.yml: [DO106] - The docker image tag is not the latest numeric tag, please update it.
The docker image tag in the yml file is: 3.10.4.30607
The latest docker image tag in docker hub is: 3.10.5.31928

**Comment by duchatran** [comment]
Is it normal for Validation tests to take 10+ hours and still going?

**Comment by kgal-pan** [comment]
> Is it normal for Validation tests to take 10+ hours and still going?

Looks like the validation job passed but it's not being reflected in the PR:
image

Merging.

**Pull request merged**